notebook2/notebook/notebook.py

```diff
@@ -106,6 +107,24 @@ def get_name(line):
         "containing the name of the docker image to use for worker pods.") from e
 
 
+def requires_auth(for_page = True):
```

Comment:
page sounds rather vague, I think I prefer `for_websocket`, it's clearer what's going on

Reply:
This would be used for anything that doesn't return html. I can change it, but if we add 1 more api route, which I think we'll likely end up doing (for instance cotton may want a cli tool), we'll need to change the name again. 

Is there another name you would like?

Reply:
nah